### ISSUE TITLE
DocumentChangeAction removed never being set

### ISSUE THREAD

**Issue by adamkoncz** [angular/angularfire#2457]
"@angular/core": "~9.0.7",
"@angular/fire": "^6.0.0",
"firebase": "^7.14.2",

On snapshotChanges the observable omits documents where **payload.type==='removed'**

```
this.afFirestore.collection(`items`).snapshotChanges().subscribe(docs => {
            docs.map(a => { console.log(`payload type ${a.payload.type}`) })
          })
```
When adding 2 items on the console the output is 

_payload.type added
payload.type added_ 

removing 1 the output is 
payload.type added

Expected behavior is returning "removed" type

_payload.type added
payload.type removed_  

An other maybe related issue is that snapshotChanges seems caching the results. It always returns all items not just the changes. If one item is modified it keeps the modified type in subsequent snapshotChanges even if it wasn't modified again. 

This is not consistent with firebase.firestore

**Comment by AveryFerrante** [comment]
I am experiencing something similar. When listening to a collection, it emits every single item in the collection, always with type 'added'. The collection could have 3 pre-existing documents. First emit from `snapshotChanges()` is the 3 documents with 'added' type. If I then add a new document, the emission is 4 documents all with 'added' type...


"firebase": "^7.9.3",
"@angular/fire": "^5.4.2",
"@angular/core": "~8.2.14"

**Comment by jamesdaniels** [comment]
I've done a lot of work on the latest release in terms of the actions. I believe this all to be addressed now in 6.1.2. Can you confirm?

**Comment by civicBoy13** [comment]
This is not fixed, im on version 6.1.5 all documentchangeaction types are still added when a new document is added

**Comment by CiriousJoker** [comment]
@jamesdaniels 
This still isn't fixed in 6.1.5, `"added"` & `"modified"` work as advertised, but `"removed"` is never set.

Checked via this code:

```ts
.snapshotChanges()
.pipe(
  map(changes => {
    const removed = changes.filter(change => change.type === "removed");
    console.log(removed.length);
  })
);
```

Here's my quickfix in the meantime:

```ts
startWith({} as ModelWrapper),
pairwise(),
map(([before, after]) => {
  const keysRemoved = Object.keys(before).filter(key => Object.keys(after).indexOf(key) < 0);

  const ret = { ...after };
  for (const keyRemoved of keysRemoved) {
    ret[keyRemoved] = before[keyRemoved];
    ret[keyRemoved].type = "removed";
  }
  return ret;
}),
```

```ts
interface ModelWrapper {
  [id: string]: DocumentChangeAction<MyModel>;
}
```

The query itself is nothing special, it just returns the whole collection desc-sorted by a date field.
